temporalio/workflow.py

```diff
@@ -5074,3 +5152,71 @@ def _to_proto(self) -> temporalio.bridge.proto.common.VersioningIntent.ValueType
         elif self == VersioningIntent.DEFAULT:
             return temporalio.bridge.proto.common.VersioningIntent.DEFAULT
         return temporalio.bridge.proto.common.VersioningIntent.UNSPECIFIED
+
+
+# Nexus
+
+ServiceT = TypeVar("ServiceT")
+
+
+class NexusClient(Generic[ServiceT]):
+    def __init__(
+        self,
+        service: Union[Type[ServiceT], str],
+        *,
+        endpoint: str,
+    ) -> None:
+        # If service is not a str, then it must be a service interface or implementation
+        # class.
+        if isinstance(service, str):
+            self._service_name = service
+        # TODO(preview): make double-underscore attrs private to nexusrpc and expose getters/setters
+        elif service_defn := getattr(service, "__nexus_service__", None):
```

Comment:
:+1: Any user should be able to obtain the service definition from a type, e.g. `nexusrpc.ServiceDefinition.from_class(service_class)`